spikeinterface/curation/remove_duplicated_spikes.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+import numpy as np
+from spikeinterface.core import BaseSorting, BaseSortingSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class RemoveDuplicatedSpikesSorting(BaseSorting):
+	"""
+	Class to remove duplicated spikes from the spike trains.
+	Spikes are considered duplicated if they are less than x
+	ms appart where x is the censored period.
+
+	Parameters
+	----------
+	sorting: BaseSorting
+		The parent sorting.
+	censored_period_ms: float
+		The censored period to consider 2 spikes to be duplicated (in ms).
+	"""
+
+	def __init__(self, sorting: BaseSorting, censored_period_ms: float = 0.3) -> None:
+		super().__init__(sorting.get_sampling_frequency(), sorting.unit_ids)
+		censored_period = int(round(censored_period_ms * 1e-3 * sorting.get_sampling_frequency()))
+
+		for segment in sorting._sorting_segments:
+			self.add_sorting_segment(RemoveDuplicatedSpikesSortingSegment(segment, censored_period, sorting.unit_ids))
+
+
+class RemoveDuplicatedSpikesSortingSegment(BaseSortingSegment):
+
+	def __init__(self, parent_segment: BaseSortingSegment, censored_period: int, unit_ids) -> None:
+		super().__init__()
+		self._parent_segment = parent_segment
+		self._duplicated_spikes = {}
+
+		for unit_id in unit_ids:
+			self._duplicated_spikes[unit_id] = self._find_duplicated_spikes(parent_segment.get_unit_spike_train(unit_id), censored_period)
+
+	@staticmethod
```

Comment:
I would putt his  as a function maybe in a separated files because this will be sued at several palces later.

Reply:
I'll create the function separately in the same file, if you want it in another file tell me which one you want :)